app/Http/Controllers/ClassController.php

```diff
@@ -70,18 +87,22 @@ public function edit(Classe $class)
     /**
      * Update the specified resource in storage.
      */
+    
     public function update(Request $request, Classe $class)
     {
         $validatedData = $request->validate([
             'className' => 'string|max:50',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
             'capacity' => 'integer',
             'price' => 'decimal:0,3',
-            
             'timeTo' => '|after:timeFrom',
         ]);
        $validatedData['isFulled']=isset($request->isFulled);
        $validatedData['timeFrom']=$request->timeFrom;
-        // dd($validatedData);
+       // dd($validatedData);
+       $image_name=$request->hasFile('image')?$this->upload_image($request->image):$request->old_image;
```

Comment:
oh you made it the two ways😂

Reply:
there is another solution we will discuss in the session, but you did a pretty good work